Document checker updates for the EU AI Act

nrichers opened the pull request:
# Pull Request Description

## What and why?
<!-- Briefly describe what changed and why — include behavior before and after if helpful -->
<!-- Frontend: Add screenshots or videos showing before/after -->

This PR:

- Adds the EU AI Act to the list of available regulations in our docs for the document checker
- Updates the same docs to mention the color badges now output by the document checker
- Updates a screenshot that shows the regulation dropdown to reflect the new option

Relates to:

- [sc-11544](https://app.shortcut.com/validmind/story/11544/add-dropdown-option-for-eu-ai-act-in-vc2-and-link-curated-questions)
- [sc-11462](https://app.shortcut.com/validmind/story/11462/documentation-for-validchecker-updates-in-25-09#activity-11740)

**Output changes**

<img width="1386" height="2136" alt="Capto_ 2025-08-27_02-30-21_pm" src="https://github.com/user-attachments/assets/c20d8dd1-fd02-47e2-916a-b4f89f34db95" />


## How to test
<!-- Describe how the change has been tested, and how a reviewer can test the change locally -->

Review the screenshot or check the preview: [**LIVE PREVIEW**](https://docs-staging.validmind.ai/pr_previews/nrichers/sc-11882/super-minor-docs-eu-ai-act-in-document-checker/guide/model-documentation/check-documentation-for-compliance.html#check-model-documentation)

## What needs special review?
<!-- Optional: Call out specific areas needing detailed review -->

N/A

## Dependencies, breaking changes, and deployment notes
<!-- A list of links to pull requests that are depend on or are dependencies of this PR -->
<!-- Any special deployment considerations? -->
<!-- List any breaking changes -->

N/A

## Release notes
<!--- REPLACE THIS COMMENT WITH YOUR DESCRIPTION --->

<!--
PR instructions for release notes:

1. Pick at least one label:

- `internal` (skip Step 2, no release notes required)
- `highlight`
- `enhancement`
- `bug`
- `breaking-change`
- `deprecation`
- `documentation`
- `environment-variables`

2. In the next section, describe the changes so that an external user can understand them. Keep it simple and link to the docs with [Learn more ...](<relative-link>), if available.
-->

## Checklist
- [x] What and why
- [x] Screenshots or videos (Frontend)
- [x] How to test
- [ ] What needs special review
- [ ] Dependencies, breaking changes, and deployment notes
- [x] Labels applied
- [x] PR linked to Shortcut
- [ ] Unit tests added (Backend)
- [x] Tested locally
- [x] Documentation updated (if required)
- [ ] Environment variable additions/changes documented (if required)

juanmleng commented:
I noticed that the example screenshot under "Each question includes:" does not show the compliance badge, and the findings and recommendations are less explicit—so it may be from a previous version. I’m sharing some updated screenshots for each compliance level here, in case you’d like to replace it with these from the newer version.

<img width="700" height="274" alt="Screenshot 2025-08-28 at 09 01 11" src="https://github.com/user-attachments/assets/97e583f9-9d5e-475f-b604-e030ccf1a923" />
<img width="698" height="286" alt="Screenshot 2025-08-28 at 09 03 15" src="https://github.com/user-attachments/assets/d6861735-464e-42bb-8813-96c2e3f78258" />
<img width="694" height="311" alt="Screenshot 2025-08-28 at 09 04 20" src="https://github.com/user-attachments/assets/50b65646-5101-48c9-b0be-235cbaca4bf3" />

validbeck commented:
Can I suggest that instead of the screenshot of the regulations, we just list the regulations available? 

This allows us to footnote links to the regulations, and when this feature is updated to accommodate custom uploads, the image also won't be inaccurate/stale/can have a footnote to the guide where you manage regulations.

nrichers commented:
> Can I suggest that instead of the screenshot of the regulations, we just list the regulations available?

Thank you for the feedback, @validbeck. 

- We already list the regulations, so no change is needed.
- Links to the regulations should not be required to use the checker, though I did try adding them. The horizontal lines for the .attn section interfered (see screenshot), so I removed them again.
- The purpose of the screenshot with the regulations is to show part of the document checker interface, since it is a feature highlight. I updated the image to better show where you invoke it from. 

I expect we will continue updating the document checker docs as we expand the feature, so nothing unusual there. 

Horizontal lines interfering: 

<img width="856" height="660" alt="Capto_ 2025-08-28_04-41-30_pm" src="https://github.com/user-attachments/assets/be227137-9e76-4046-a2a4-5638d593f727" />
